app/controllers/rooms_controller.rb

```diff
@@ -115,41 +115,65 @@ def game_player_count_post
     count = params[:player_count].to_i
     @room.player_count = count
     @room.save
+    # Delete players that no longer fit in this game
+    ActiveRecord::Base.connection.execute("DELETE FROM room_players where room_id = #{@room.id} AND player_number > #{(count / 2).to_i}")
```

Comment:
I have an answer!

```ruby
irb(main):002:0> Room.new(client_token: "qwer", name: "fafa").save
   (0.1ms)  begin transaction
  SQL (0.7ms)  INSERT INTO "rooms" ("client_token", "created_at", "name", "updated_at") VALUES (?, ?, ?, ?)  [["client_token", "qwer"], ["created_at", "2016-04-13 17:26:26.683462"], ["name", "fafa"], ["updated_at", "2016-04-13 17:26:26.683462"]]
   (0.7ms)  commit transaction
=> true
irb(main):003:0> Room.all.where('client_token = ? AND name = ?', 'qwer', 'fafa').delete_all
  SQL (1.4ms)  DELETE FROM "rooms" WHERE (client_token = 'qwer' AND name = 'fafa')
=> 1
irb(main):004:0> Room.new(client_token: "qwer", name: "fafa").save
   (0.1ms)  begin transaction
  SQL (0.3ms)  INSERT INTO "rooms" ("client_token", "created_at", "name", "updated_at") VALUES (?, ?, ?, ?)  [["client_token", "qwer"], ["created_at", "2016-04-13 17:27:13.269485"], ["name", "fafa"], ["updated_at", "2016-04-13 17:27:13.269485"]]
   (1.4ms)  commit transaction
=> true
irb(main):005:0> Room.all.where('client_token = ? AND name = ?', 'qwer', 'fafa').destroy_all
  Room Load (0.2ms)  SELECT "rooms".* FROM "rooms"  WHERE (client_token = 'qwer' AND name = 'fafa')
   (0.0ms)  begin transaction
  SQL (0.3ms)  DELETE FROM "rooms" WHERE "rooms"."id" = ?  [["id", 3]]
   (1.5ms)  commit transaction
=> [#<Room id: 3, client_token: "qwer", team_a_score: nil, team_b_score: nil, name: "fafa", created_at: "2016-04-13 17:27:13", updated_at: "2016-04-13 17:27:13", game: nil>]
```

So, when you do `#delete_all`, it will not hydrate! it just constructs the delete statement and fires it off.